pkg/reconciler/labeler/labels.go

```diff
@@ -46,16 +46,30 @@ func (c *Reconciler) syncLabels(ctx context.Context, r *v1.Route) error {
 	configs := sets.NewString()
 
 	// Walk the revisions in Route's .status.traffic and build a list
-	// of Configurations to label from their OwnerReferences.
+	// of RunLatest Configurations to label from their OwnerReferences.
 	for _, tt := range r.Status.Traffic {
 		rev, err := c.revisionLister.Revisions(r.Namespace).Get(tt.RevisionName)
 		if err != nil {
 			return err
 		}
 		revisions.Insert(tt.RevisionName)
+
+		// If the owner reference is a configuration, add it to the list of configurations
+		// and add its LatestCreatedRevision to the list of revisions
 		owner := metav1.GetControllerOf(rev)
 		if owner != nil && owner.Kind == "Configuration" {
 			configs.Insert(owner.Name)
+
+			config, err := c.configurationLister.Configurations(r.Namespace).Get(owner.Name)
+			if err != nil {
+				return err
+			}
+
+			// If we are tracking the latest revision, we need to label the latest created revision
+			// as well so that there is a smooth transition when the new revision becomes ready.
+			if tt.LatestRevision != nil && *tt.LatestRevision {
+				revisions.Insert(config.Status.LatestCreatedRevisionName)
```

Comment:
@vagababov @markusthoemmes it is tempting to also label the LCR with the LRR in this case so that the KPA gets a hint about how big to make the LCR before it is receiving traffic 🤔

Reply:
(not for this PR)